messaging/repository/room_repository.go

```diff
@@ -0,0 +1,210 @@
+
+	addRoomForParticipant = `UPDATE chat.student_rooms  SET rooms = rooms +? WHERE student=? ;`
+	CreateStudentRooms=`INSERT INTO chat.student_rooms (student) VALUES (?);`
+	RemoveRoomForParticipant = `UPDATE chat.student_rooms  SET rooms = rooms-? WHERE student=? ;`
+
+)
+
+
+func (r RoomRepository) SaveRoom( room *domain.ChatRoom) error {
+	var students []string
+	for _, element := range room.Students{
+		students = append(students, element.ID)
+	}
+
+	 err := r.dbSession.Query(SaveRoom, room.RoomID, room.Admin.ID,room.Name,students).Consistency(gocql.One).Exec();
+
+	if err!=nil {
+		log.Fatal(err)
+	}
+
+
+	fmt.Println("Message entry: ", room.RoomID, room.Admin,room.Name,room.Students)
+
+	return err
+
+}
+
+
+func (r RoomRepository) GetRoom( roomID string) (*domain.ChatRoom, error) {
+	var room domain.ChatRoom
+
+	var studentText  []string
+	var AllStudent []domain.Student
+
+	err := r.dbSession.Query(GetRoom, roomID).Consistency(gocql.One).Scan(&room.RoomID, &room.Admin.ID,&room.Deleted, &room.Name, &studentText);
+
+
+	if err!=nil {
+		log.Fatal(err)
+		return nil,err ;
+	}
+
+	for _,ID := range studentText{
+		var student domain.Student
+		student.ID=ID
+		AllStudent = append(AllStudent,student )
+
+	}
+	room.Students= AllStudent
+
+
+
+
+
+	return &room,err
+
+}
+
+
+func (r RoomRepository) EditChatRoomParticipants( roomID string, student []domain.Student) error {
+var studentID []string
+
+	for _,s := range student{
+
+		studentID = append(studentID,s.ID )
+
+	}
+	err := r.dbSession.Query(editChatroomParticipant, studentID, roomID).Consistency(gocql.One).Exec();
+	if err!=nil {
+		log.Fatal(err)
+		return err ;
+	}
+
+	return  nil
+}
+
+func (r RoomRepository) GetRoomsFor( studentID string) (*domain.StudentChatRooms, error) {
+	var StudentRoom domain.StudentChatRooms
+	var roomsID []string
+	var rooms []domain.ChatRoom
+	err := r.dbSession.Query(GetRoomFor, studentID).Consistency(gocql.One).Scan(&StudentRoom.Student.ID, &roomsID);
+	if err!=nil {
+		log.Fatal(err)
+		return nil,err ;
+	}
+	for _,ID := range roomsID{
+		var room domain.ChatRoom
+		room.RoomID=ID
+		rooms = append(rooms,room )
+
+	}
+	StudentRoom.Rooms=rooms
+
+	return &StudentRoom , err
+}
+
+
+
+func (r RoomRepository) AddRoomForParticipants( roomID string, students []domain.Student) error {
+	var rooms []string
+	rooms=append(rooms,roomID)
+
+	for _,student:= range students {
+
+		err := r.dbSession.Query(addRoomForParticipant, rooms,student.ID).Consistency(gocql.One).Exec();
+		if err!=nil {
+			log.Fatal(err)
+			return err ;
+		}
+	}
+
+	return  nil
+
+}
+
+func (r RoomRepository) CreateStudentRooms( studnet domain.Student) error {
+
+
+
+		err := r.dbSession.Query(CreateStudentRooms, studnet.ID).Consistency(gocql.One).Exec();
+		if err!=nil {
+			log.Fatal(err)
+			return err ;
+		}
+
+
+	return  nil
+
+}
+
+
+
+func (r RoomRepository) RemoveRoomForParticipants( roomID string, students []domain.Student) error {
```

Comment:
should we check that the room exists before trying to delete it? then return a message saying room does not exist and exit normally instead of panic

Reply:
I could not add that in the repo layer , so i did that in the usecase